packages/piro/test/Main_AnalysisDriver_Tpetra.cpp

```diff
@@ -227,8 +227,8 @@ int main(int argc, char *argv[]) {
             p_exact[1] = 4;
           }
           if (mockModel=="MockModelEval_B_Tpetra_2_parameters") {
-            p_exact[0] = 6;
-            p_exact[1] = 4;
+            p_exact[0] = 4;
```

Comment:
I think you should not change this, but below use `p_view(p_indices[0]) `instead of `p_view(0)`, etc.

Reply:
> Thanks @mperego ! I am currently trying without ROL. For the extra test in `Main_ThyraSolver_Tpetra.cpp` can it be done in a different PR? It will require some time as right now this test only uses one of the `MockModelEval` (and not the one I modified).

sounds good.

Reply:
> I think you should not change this, but below use `p_view(p_indices[0]) `instead of `p_view(0)`, etc.

@mperego I am not sure about that because most of the tests tested in this executable only have 1 parameter (with 2 entries) and therefore do not have `p_indices[1]`.

So something like:
```
double l2_diff = std::sqrt(std::pow(p_view(p_indices[0])-p_exact[0],2) + std::pow(p_view(p_indices[1])-p_exact[1],2));
```
does not work for the other tests (but works for the added test of this PR).

We can modify this verification check to have two different code paths though but I am not sure if it is the best.

Reply:
OK, we can address this later.